### ISSUE TITLE
Request custom support for blob storage when emulated

### ISSUE THREAD

**Issue by wn-na** [microsoft/code-push-server#54]
hello.

I have an issue where the download URL comes out as 127.0.0.1 when emulated. 

To solve this, I am using the following method.

- `api/script/routes/acquisition.ts` add to line 182
```typescript
          if (process.env.EMULATED) {
            updateCheckBody.updateInfo.downloadURL = updateCheckBody.updateInfo.downloadURL.replace(
              "http://127.0.0.1:10000",
              "http://MY_LOACL_HOST_IP:10000"
            );
          }
```

Is it possible to officially add a custom env setting for the blob storage address?

**Comment by felixchan** [comment]
I have tried the same hack to replace 127.0.0.1 azurite URL to my own domain.
The result is that my client will download the package from is something like this.

http://api2.mydomain.com:10000/devstoreaccount1/storagev2/sWXwSg7ek5XAktIXLVNIW6nczYRzV1CPmJbV-x

However, when my client attempts to actually download the package (from my remote server, port 10000, azurite emulated), azurite returns 404

**Comment by wn-na** [comment]
@felixchan Did you give `--blobHost 0.0.0.0` option when running azurite?

**Comment by felixchan** [comment]
> [@felixchan](https://github.com/felixchan) Did you give `--blobHost 0.0.0.0` option when running azurite?

Yes, I did

When my iOS react-native client app gets the update package, it correctly lists my domain:

```
{
  description: 'menu4',
  label: 'v6',
  appVersion: '3.0',
  isMandatory: false,
  packageHash: 'c89b441277e1e0499b8617ff1b1634ea3de8d13e39f74e3f31',
  packageSize: 307082160,
  downloadUrl: 'http://api2.mydomain.com:10000/devstoreaccount1/storagev2/sWXwSg7ek5mAktIXIW6nczYRzV1CPmJbV-x',
 }
```

However, when I GET that downloadUrl, I get a 404.  

Were you able to successfully download from azurite?

**Comment by wn-na** [comment]
@felixchan yes, i can download for ios and android. can you check it for `info.plist` to allow your download domain?

**Comment by felixchan** [comment]
> [@felixchan](https://github.com/felixchan) yes, i can download for ios and android. can you check it for `info.plist` to allow your download domain?

Info.plist allows my custom domain for HTTP.

```
	<key>NSAppTransportSecurity</key>
<dict>
    <key>NSAllowsArbitraryLoads</key>
    <true/>
    <key>NSExceptionDomains</key>
    <dict>
        <key>api2.mydomain.com</key>
        <dict>
            <key>NSIncludesSubdomains</key>
            <true/>
            <key>NSTemporaryExceptionAllowsInsecureHTTPLoads</key>
            <true/>
            <key>NSTemporaryExceptionMinimumTLSVersion</key>
            <string>TLSv1.0</string>
        </dict>
    </dict>
</dict>
```

When you GET the `downloadUrl` normally,  does it download? 

It's 404 even for a normal GET.

**Comment by wn-na** [comment]
@felixchan I've been downloading just fine.

 ![Image](https://github.com/user-attachments/assets/fa1c9a57-d75c-4679-a122-0de1a9a2a935)

i run azurite for this command
`azurite --skipApiVersionCheck  --disableProductStyleUrl --blobHost 0.0.0.0 --loose`

i think 404 is maybe issue for your domain (like nginx setting or tomcat, etc...)

**Comment by felixchan** [comment]
@wn-na I was able to download the package after applying your patch to azure-storage. (Before , I forgot a step)

Thanks so much!

**Comment by KarthikeyanRaju-T2S** [comment]
@wn-na  I have followed the same, it's working expected :) thanks. only thing is it's not showing the `Install Metrics`, like it's always shows `No installs recorded`

**Comment by eithe** [comment]
@KarthikeyanRaju-T2S For metrics you would need to [setup Redis as explained in the Readme](https://github.com/microsoft/code-push-server/blob/main/api/README.md#metrics).

**Comment by KarthikeyanRaju-T2S** [comment]
@eithe I have install redis in the ubuntu instance(where I have installed and running the code-push-server) also updated the .env with `REDIS_HOST` and `REDIS_PORT, after this when I try to access the Apps to see the release history it's throwing error `[Error]  error code: 504`

Since redis is in the same machine where code-push-server is setup and running, I didn't configure the TLS for redis.
